### PR TITLE
Add Kubernetes test using kind

### PR DESCRIPTION
**Pull request by jawnsy**
Closes https://github.com/PrefectHQ/platform/issues/2922

### Checklist

- [x] References any related issue by including "Closes #<Issue Number>" or "Closes <Issue URL>".
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect-kubernetes/issues/new/choose) first.
- [ ] Includes tests or only affects documentation.
- [ ] Passes `pre-commit` checks.
  - Run `pre-commit install && pre-commit run --all` locally for formatting and linting.
- [ ] Includes screenshots of documentation updates.
  - Run `mkdocs serve` view documentation locally.
- [ ] Summarizes PR's changes in [CHANGELOG.md](https://github.com/PrefectHQ/prefect-kubernetes/blob/main/CHANGELOG.md)

### CODE REVIEW

**Review by madkinsz — APPROVED** [review]
Siiick

File: .github/workflows/kubernetes-test.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        run: kubectl logs deployment/prefect-server
+
+      - name: Install worker chart and override image
+        run: |
+          helm install prefect-worker prefect/prefect-worker \
+            --set=worker.apiConfig=server \
+            --set=worker.serverApiConfig.apiUrl=http://prefect-server.default.svc.cluster.local:4200 \
+            --set=worker.image.repository=localhost/kubernetes-worker \
+            --set=worker.image.prefectTag=latest \
+            --wait
```
Comment (desertaxle) [inline .github/workflows/kubernetes-test.yml:68]: We'll need to set a work pool for the worker to start up without an error. I think this is the correct way to do that?
```suggestion
          helm install prefect-worker prefect/prefect-worker \
            --set=worker.apiConfig=server \
            --set=worker.serverApiConfig.apiUrl=http://prefect-server.default.svc.cluster.local:4200 \
            --set=worker.image.repository=localhost/kubernetes-worker \
            --set=worker.image.prefectTag=latest \
            --set=worker.config.workPool=practice \
            --wait
```

**Comment by desertaxle** [comment]
This is really cool! I think once we're setting the work pool for the worker we can merge this and explore executing flow runs with the worker in subsequent PRs.

**Review by desertaxle — APPROVED** [review]
Sweet!

**Pull request merged**